src/charm.py

```diff
@@ -128,9 +128,9 @@ def __init__(self, *args):
                 config=to_tuple(ActiveStatus()),
                 rules=to_tuple(ActiveStatus()),
                 retention=to_tuple(ActiveStatus()),
-            )
+            ),
+            fresh_install=True,
         )
-        self._stored.set_default(fresh_install=True)  # Relies on controller-backed storage
 
```

Comment:
@IbraAoad: we initially said this relies on controller storage, but now that I see we set `fresh_install` it to False and the config backup is saved to persistent storage, it seems that controller storage isn't necessary. Unless I'm missing something?

Reply:
yeah I think I agree with this.  Persisting data between pods due to upgrade, churn, etc, shouldn't be needed.  This flag I think is only meaningful within a given pod's lifecycle.  It says whether this pod started its life in an install or upgrade, and I think that's all that's necessary